packages/eventual-send/src/index.js

```diff
@@ -224,13 +236,24 @@ export function makeHandledPromise(Promise) {
     return handledP;
   }
 
-  HandledPromise.prototype = Promise.prototype;
+  HandledPromise.prototype = promiseProto;
 
   // Uncomment this line if needed for conformance to the proposal.
   // Currently the proposal does not specify this, but we might change
   // our mind.
   // Object.setPrototypeOf(HandledPromise, Promise);
 
+  function isNormalPromiseThen(p) {
+    return (
+      (!isSES || isFrozen(p)) &&
+      // isFrozen(promiseProto) && // unnecessary under SES and too strict outside SES
+      getPrototypeOf(p) === promiseProto &&
+      promiseResolve(p) === p &&
+      gopd(p, 'then') === undefined &&
+      (isSES || gopd(promiseProto, 'then').value === originalThen) // unnecessary under SES
```

Comment:
Outside of SES, if we're not checking `isFrozen(promiseProto)` and we are checking `gopd(promiseProto, 'then').value === originalThen`, we should also check that this data property is non-writable, non-configurable.

Reply:
I found that using `harden()` anywhere in `eventual-send` meant that we are taking a stand in hardening HandledPromises.  Going all the way, I instead harden things that are returned by  `HandledPromise.resolve` which simplifies the test and makes it behave more consistently inside and outside SES.  This helped ERTP pass all its unit tests again.

I hope I made an acceptable tradeoff.

If that's not acceptable, then we need a much more convincing story of what we should actually harden in `eventual-send`.

Reply:
@erights PTAL.